src/main/java/subway/repository/DbLineRepository.java

```diff
@@ -0,0 +1,157 @@
+package subway.repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Repository;
+import subway.dao.LineDao;
+import subway.dao.StationEdgeDao;
+import subway.domain.Line;
+import subway.domain.StationEdge;
+import subway.entity.LineEntity;
+import subway.entity.StationEdgeEntity;
+import subway.exception.StationEdgeNotFoundException;
+
+
+@Repository
+public class DbLineRepository implements LineRepository {
+
+    private final LineDao lineDao;
+    private final StationEdgeDao stationEdgeDao;
+
+
+    public DbLineRepository(LineDao lineDao, StationEdgeDao stationEdgeDao) {
+        this.lineDao = lineDao;
+        this.stationEdgeDao = stationEdgeDao;
+    }
+
+    @Override
+    public Optional<Line> findById(Long id) {
+        Optional<LineEntity> optionalLineEntity = lineDao.findById(id);
+        if (optionalLineEntity.isEmpty()) {
+            return Optional.empty();
+        }
+
+        LineEntity lineEntity = optionalLineEntity.get();
+
+        return convertToDomain(lineEntity);
+    }
```

Comment:
```suggestion
        return lineDao.findById(id)
                .flatMap(this::convertToDomain);
    }
```
convertToDomain이 Line 을 반환한다면 map이어도 되겠네요 :)

Reply:
코즈 짱!! 덕분에 이번에 옵셔널을 다시 학습해보았슴다 ! 옵셔널 맵을 이렇게 쓰는 것이군요!!